src/Estimators/SelfHealingOverlap.cpp

```diff
@@ -0,0 +1,162 @@
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2021 QMCPACK developers.
+//
+// File developed by: Jaron T. Krogel, krogeljt@ornl.gov, Oak Ridge National Laboratory
+//
+//////////////////////////////////////////////////////////////////////////////////////
+
+#include "SelfHealingOverlap.h"
+#include "TrialWaveFunction.h"
+#include "QMCWaveFunctions/Fermion/MultiSlaterDetTableMethod.h"
+
+#include <iostream>
+#include <numeric>
+
+
+namespace qmcplusplus
+{
+SelfHealingOverlap::SelfHealingOverlap(SelfHealingOverlapInput&& inp_,
+                                       const TrialWaveFunction& wfn,
+                                       DataLocality dl)
+  : OperatorEstBase(dl),
+    input_(std::move(inp_))
+{
+
+  //my_name_ = input_.get_name();
+
+  auto& inp = this->input_.input_section_;
+
+  auto msd_or_null = wfn.findMSD();
+  if(!msd_or_null)
+    throw std::runtime_error("SelfHealingOverlap only supports multi slater determinant wavefunctions which cannot be found in the trial wavefunction.");
+
+  const MultiSlaterDetTableMethod& msd = *msd_or_null;
+
+  // JTK: how to get # of coeff in multidet at this point?
+  //size_t data_size = 1486;
+  const size_t data_size = msd.getLinearExpansionCoefs().size();
+  data_.resize(data_size, 0.0);
+}
+
+
+
+SelfHealingOverlap::SelfHealingOverlap(const SelfHealingOverlap& sh, DataLocality dl) : SelfHealingOverlap(sh)
+{
+  data_locality_ = dl;
+}
+
+std::unique_ptr<OperatorEstBase> SelfHealingOverlap::spawnCrowdClone() const
+{
+  std::size_t data_size    = data_.size();
+  auto spawn_data_locality = data_locality_;
+
+  if (data_locality_ == DataLocality::rank)
+  {
+    // This is just a stub until a memory saving optimization is deemed necessary
+    spawn_data_locality = DataLocality::queue;
+    data_size           = 0;
+    throw std::runtime_error("There is no memory savings implementation for SelfHealingOverlap");
+  }
+
+  auto spawn = std::make_unique<SelfHealingOverlap>(*this, spawn_data_locality);
+  spawn->get_data().resize(data_size);
+  return spawn;
+}
+
+void SelfHealingOverlap::startBlock(int steps)
+{
+}
+
+/** Gets called every step and writes to thread local data.
+ *
+ */
+void SelfHealingOverlap::accumulate(const RefVector<MCPWalker>& walkers,
+                                    const RefVector<ParticleSet>& psets,
+                                    const RefVector<TrialWaveFunction>& wfns,
+                                    const RefVector<QMCHamiltonian>& hams,
+                                    RandomBase<FullPrecRealType>& rng)
+{
+  for (int iw = 0; iw < walkers.size(); ++iw)
+  {
+    MCPWalker& walker      = walkers[iw];
+    ParticleSet& pset      = psets[iw];
+    TrialWaveFunction& psi = wfns[iw];
+    RealType weight        = walker.Weight;
+    auto& wcs              = psi.getOrbitals();
```

Comment:
Not for this PR but we should change the name of this function. It should not be call 'orbitals.'